python/pyspark/streaming/tests.py

```diff
@@ -408,13 +408,17 @@ def test_failed_func(self):
         input_stream = self.ssc.queueStream(input)
 
         def failed_func(i):
```

Comment:
There should be a few tests to make sure that reusing the same python object always resets the failure.